airsonic-main/src/main/java/org/airsonic/player/spring/SonosConfiguration.java

```diff
@@ -0,0 +1,37 @@
+package org.airsonic.player.spring;
+
+import org.airsonic.player.service.SonosService;
+import org.airsonic.player.service.sonos.SonosFaultInterceptor;
+import org.airsonic.player.service.sonos.SonosHelper;
+import org.apache.cxf.jaxws.EndpointImpl;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.ImportResource;
+
+import javax.xml.ws.Endpoint;
+
+import java.util.Collections;
+
+@Configuration
+@ImportResource({"classpath:META-INF/cxf/cxf.xml", "classpath:META-INF/cxf/cxf-servlet.xml"})
+public class SonosConfiguration {
+
+    @Bean
+    public SonosFaultInterceptor sonosFaultInterceptor() {
+        return new SonosFaultInterceptor();
+    }
+
+    @Bean
+    public Endpoint sonosEndpoint(SonosService sonosService) {
+        EndpointImpl endpoint = new EndpointImpl(sonosService);
+        endpoint.publish("/Sonos");
+        endpoint.setOutFaultInterceptors(Collections.singletonList(sonosFaultInterceptor()));
+        return endpoint;
+    }
+
+    @Bean
+    public SonosHelper sonosHelper() {
+        return new SonosHelper();
```

Comment:
This is likely a bug. You've instantiated and returned a blank SonosHelper here with no dependencies wired inside.

You actually don't need to, SonosHelper is already annotated with a component (@Service) so it will autoinstantiate, therefore this bean is not needed

Reply:
Done.